alica_tests/src/test/test_blackboard.cpp

```diff
@@ -84,7 +84,7 @@ TEST_F(SingleAgentTestFixture, testValueMappingBehaviours)
     ASSERT_TRUE(_tc->setTransitionCond("BlackboardTestPlan", "ChooseBlackboardTestState", "ValueMappingBehaviourTestState")) << _tc->getLastFailure();
 
     // Step until the plan succeeds
-    UnlockedBlackboard gb(_tc->editGlobalBlackboard());
+    LockedBlackboardRO gb(_tc->editGlobalBlackboard());
```

Comment:
You are holding lock while stepping the engine, so the value will never be able to change

Reply:
True, when tests start failing this will cause issues. I will revert it back to the UnlockedBlackboard

Reply:
Proper behavior would be to acquire the lock every time you check the value